common/markdownPreview.js

```diff
@@ -0,0 +1,90 @@
+/**
+* @desc
+* The markdownPreview directive can be used to display a modal preview box with
+* transformed markdown. It calls ERMrest.renderMarkdown() to perform markdown to
+* HTML transformtaion. The output of this public function is then bound to page
+* through ng-bind-html.
+* The markdownPreview directive can be used as an element tag (<markdown-preview></markdown-preview>)
+* or an attribute (<div markdown-preview></div>). Link function checks for class "live-preview" to identify
+* click event, therefore this should be added in html markup.
+* It accepts the following attribute:
+* @param {String} textinput The Markdown to transform
+* @example <a href class="live-preview"  markdown-preview textinput="form.recordEditModel.rows[rowIndex][column.name]">Live Preview</a>
+* @example <markdown-preview class="live-preview" textinput="form.recordEditModel.rows[rowIndex][column.name]">Click</markdown-preview>
+*/
+
+(function() {
+  'use strict';
+
+  angular.module('chaise.markdown', [])
+    .directive('markdownPreview', ['$uibModal', 'ERMrest', '$sce', function($uibModal, ERMrest, $sce) {
+
+      return {
+        restrict: 'EA',
+        scope: {
+          textinput: '=textinput'
+        },
+        link: function(scope, elem, attr) {
+
+          elem.on('click', function(e) {
+            scope.$apply(function() {
+              if (angular.element(e.target).hasClass('live-preview')) {
+                var result;
+                var textInput = scope.textinput;
+
+                angular.isUndefinedOrNull = function(val) {
+                  return val == '' || angular.isUndefined(val) || val === null
+                }
+
+                function modalBox(params) {
+                  var modalInstance = $uibModal.open({
+                    animation: false,
+                    controller: "MarkdownPreviewController",
+                    controllerAs: "ctrl",
+                    resolve: {
+                      params: params
+                    },
+                    size: "lg",
+                    template: '<div class="modal-header"> \
```

Comment:
Is there a reason that this template isn't defined in an html file (common/templates/markdown-preview.html)?